service/controller/key/generate/generate_amis.go

```diff
@@ -0,0 +1,110 @@
+//go:generate go run generate_amis.go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"golang.org/x/net/html"
+
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+const keyPackage = "key"
+const filename = "amis.go"
+const sourceFileTemplate = `package {{ .Package }}
+
+import "encoding/json"
+
+var amiJSON = []byte({{ .AMIInfoString }})
+var amiInfo = map[string]AMIInfoList{}
+
+func init() {
+	err := json.Unmarshal(amiJSON, &amiInfo)
+	if err != nil {
+		panic(err)
+	}
+}
+`
+
+type sourceFileTemplateData struct {
+	AMIInfoString string
+	Package       string
+}
+
+func main() {
+	url := "https://stable.release.core-os.net/amd64-usr/"
+	response, err := http.Get(url)
+	if err != nil {
+		log.Fatal(err)
+	}
+	z := html.NewTokenizer(response.Body)
+
+	var versions []string
+	end := false
+	for !end {
+		tt := z.Next()
+		switch {
+		case tt == html.ErrorToken:
+			end = true
+		case tt == html.StartTagToken:
+			t := z.Token()
+			if t.Data == "a" {
+				versions = append(versions, strings.TrimSuffix(t.Attr[0].Val, "/"))
+			}
+		}
+	}
+
+	versionAMIs := map[string]key.AMIInfoList{}
+	for _, version := range versions {
+		url := fmt.Sprintf("https://stable.release.core-os.net/amd64-usr/%s/coreos_production_ami_all.json", version)
```

Comment:
Just thinking: We will need to change this URL eventually when `core-os` finally goes the way of to dodo.
We should make this configurable through e.g. `installations` because otherwise it will be a very sad day when we need to scramble and patch old operator versions which stopped working because the URL does not serve anymore.

Reply:
Okay I just saw that this is fixed in a PR on this branch - there are too many branches :D